src/main/java/com/github/eyefloaters/console/api/model/RecordFilterParams.java

```diff
@@ -3,48 +3,53 @@
-import jakarta.validation.constraints.Min;
-import jakarta.validation.constraints.Positive;
 import jakarta.ws.rs.DefaultValue;
 import jakarta.ws.rs.QueryParam;
 
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
 
-import com.github.eyefloaters.console.legacy.model.Types.Record;
+import com.github.eyefloaters.console.api.support.ErrorCategory;
+import com.github.eyefloaters.console.api.support.RangeLimitedInteger;
 
 public class RecordFilterParams {
 
-    public static final String PROP_LIMIT = "limit";
-    public static final String PROP_MAX_VALUE_LENGTH = "maxValueLength";
-
-    @QueryParam(Record.PROP_PARTITION)
+    @QueryParam("filter[partition]")
     @Parameter(description = "Retrieve messages only from this partition")
-    Integer partition;
-
-    @QueryParam(Record.PROP_OFFSET)
-    @Parameter(description = "Retrieve messages with an offset equal to or greater than this offset. If both `timestamp` and `offset` are requested, `timestamp` is given preference.")
-    @Min(0)
-    Integer offset;
-
-    @QueryParam(Record.PROP_TIMESTAMP)
+    @RangeLimitedInteger(min = "0", category = ErrorCategory.INVALID_QUERY_PARAMETER, source = "filter[partition]")
+    String partition;
+
+    @QueryParam("filter[offset]")
+    @Parameter(description = """
+            Retrieve messages with an offset equal to or greater than this offset.
+            If both `filter[timestamp]` and `filter[offset]` are requested, `filter[timestamp]`
+            is given preference.
+            """)
+    @RangeLimitedInteger(min = "0", category = ErrorCategory.INVALID_QUERY_PARAMETER, source = "filter[offset]")
+    String offset;
+
+    @QueryParam("filter[timestamp]")
     @Parameter(
-        description = "Retrieve messages with a timestamp equal to or later than this timestamp. If both `timestamp` and `offset` are requested, `timestamp` is given preference.",
+        description = """
+            Retrieve messages with a timestamp equal to or later than this timestamp.
+            If both `filter[timestamp]` and `filter[offset]` are requested, `filter[timestamp]`
+            is given preference.
+            """,
         schema = @Schema(format = "date-time"))
     String timestamp;
 
-    @QueryParam(PROP_LIMIT)
+    @QueryParam("page[size]")
     @DefaultValue("20")
     @Parameter(description = "Limit the number of records fetched and returned")
-    @Positive
-    Integer limit;
+    @RangeLimitedInteger(category = ErrorCategory.INVALID_QUERY_PARAMETER, source = "page[size]")
+    String limit;
 
-    @QueryParam(PROP_MAX_VALUE_LENGTH)
+    @QueryParam("maxValueLength")
     @Parameter(description = "Maximum length of string values returned in the response. "
             + "Values with a length that exceeds this parameter will be truncated. When this parameter is not "
             + "included in the request, the full string values will be returned.")
```

Comment:
This parameter came from RHOSAK where the UI lists a truncated value in the grid. As you figured, it was to prevent fetching large records for no reason. I believe the UI would fetch again without the limit to display the value for a single record.

I can add something to the docs saying that values returned when this parameter is used should assume a value length equal to the param value means the value was truncated. We could also perhaps use the `meta` field map [1] to indicate that a value was truncated.

[1] https://jsonapi.org/format/#document-meta